agents/baseAgent.go

```diff
@@ -0,0 +1,63 @@
+package agents
```

Comment:
every new package should have a doc.go file (I need to write up contribution guidelines, I will soon).